File: problems/0128-longest-consecutive-sequence/solution.cpp
#include<algorithm>

class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        std::sort(nums.begin(), nums.end(), [](int a, int b) { return a < b; });

        if (nums.size() <= 1) return nums.size();

        int longest = 1;
        int consec_length = 1;
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] == nums[i - 1]) continue;
            if (nums[i] == nums[i - 1] + 1) {
                consec_length++;
            } else {
                consec_length = 1;
            }
            if (consec_length > longest) {
                longest = consec_length;
            }
        }
        return longest;
    }
};

File: problems/0128-longest-consecutive-sequence/solution.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nset = set(nums)
        ml = 0

        for n in nset:
            if n - 1 not in nset:
                i = 0
                while n + i in nset:
                    i += 1
                ml = max(ml, i)
        
        return ml
